### PR TITLE
feat(base-entity): fields and ids getters on base entity

### PR DESCRIPTION
**Pull request by euduardo**
A clever and helpful way to expose the fields and ids of an entity

Resolves the issue #60 

## Proposed Changes

1. Created 2 static getters on base entity: `fields` and `ids`
2. `fields` exposes the metadata of the entity fields
3. `ids` exposes the metadata of the entity ids (fields created with `id` ou with `idId = true` option)

### Author/Contributor
- [x] If documentation is needed for this change, has that been included in this pull request
- [x] Remember to check if code coverage decrease, if so, please implement the necessary tests

### Reviewing Maintainer
- [ ] Label as `breaking` if this is a large fundamental change
- [ ] Label as either `automation`, `bug`, `documentation`, `enhancement`, `infrastructure`, or `performance`

### CODE REVIEW

**Review by jhomarolo — CHANGES_REQUESTED** [review]
Hi @euduardo thanks for your contribution!

I think is missing some documentation about how to use this.

**Comment by dalssoft** [comment]
great contribution. tks! I just think we should discuss a bit more regarding the DX.

with the current implementation, if I got it right, would be possible to do something like:

`const ids = MyEntity.ids    // list of id names`

My concern is that it does not make it explicit that it is a meta data info and it can mess with the fields from the entity. So my suggestion would be:

`const ids = MyEntity.schema.ids`

So for every schema metadata info we could use the `schema` property (ex: `ids`, `fields`, etc).

Another important question is how `MyEntity.schema` would differ from `prototype.meta.schema`?

**Comment by euduardo** [comment]
> great contribution. tks! I just think we should discuss a bit more regarding the DX.
> 
> with the current implementation, if I got it right, would be possible to do something like:
> 
> `const ids = MyEntity.ids // list of id names`
> 
> My concern is that it does not make it explicit that it is a meta data info and it can mess with the fields from the entity. So my suggestion would be:
> 
> `const ids = MyEntity.schema.ids`
>
> So for every schema metadata info we could use the `schema` property (ex: `ids`, `fields`, etc).

My intention with this PR is to exposing those things in a centralized API, so some open issues (like https://github.com/herbsjs/herbs2rest/issues/21) could take advantage of it. 

Since those fields are static getters, they would be accessible through the class, not through the instance. I thought that would be different enough, but I agree with you that an unwary user could struggle with a mistake.
 
> Another important question is how `MyEntity.schema` would differ from `prototype.meta.schema`?

Good point. Creating another `schema` proeprty could be event more confusing. 

I can think in some different approaches to solve this:

1) Using a prefix (`$`, `_`, or even `__`) in those fields:
```javascript
const entityIds = MyEntity.$ids
// or
const entityIds = MyEntity._ids
// or
const entityIds = MyEntity.__ids
```

2) Using a helper function to access that kind of property:
```javascript
const { entityMetaReader } = require('@herbsjs/gotu') // I coulnd't find a good name
const entityIds = entityMetaReader(MyEntity).ids
```

3) Creating those things on `Entity.prototype.meta`:
```javascript
const entityIds = MyEntity.prototype.meta.schema.ids
// or
const entityIds = MyEntity.prototype.meta.ids
```

What are your thoughs?

**Comment by dalssoft** [comment]
I like the general idea. Is that ok if we go with `MyEntity.schema.xyz` as a shortcut for `MyEntity.prototype.meta.schema`?

**Comment by euduardo** [comment]
> I like the general idea. Is that ok if we go with `MyEntity.schema.xyz` as a shortcut for `MyEntity.prototype.meta.schema`?

Properties updated.

File: README.md [hunk 1]
```diff
@@ -366,6 +366,71 @@ Check if an object is a Gotu Entity class
         entity.isEntity(Object) // false
 ```
 
+## Entity schema - `entity.schema`
+
+It is possible to access the schema information of your entity. There are two special properties:
+
+### fields
+
+Returns an list with all the fields of the entity:
```
Comment (jhomarolo) [inline README.md:365]: Would be good to add how isID works here and change the documentation in that section.

image

**Comment by dalssoft** [comment]
It seems the only thing missing to merge is the updated site doc, right?

File: README.md (hunk 1 above)
Comment (euduardo) [reply README.md:365]: Done!

**Comment by dalssoft** [comment]
Is there an open issue to fix/update herbs site/doc?

**Comment by eduardo-vortx** [comment]
> Is there an open issue to fix/update herbs site/doc?

No, there is not.

**Comment by dalssoft** [comment]
The best practice would be to create a issue there and send a PR before merge this current PR.

https://github.com/herbsjs/herbsjs.github.io/issues

**Comment by italojs** [comment]
I think in this case isn't necessary to create an issue because we already had a long discussion here, just open the PR and link this PR discussion.

Great contribution @euduardo  🎊

**Comment by dalssoft** [comment]
The issue for the doc is here: https://github.com/herbsjs/herbsjs.github.io/issues/120

Sorry guys, I'm not merging this PR without docs. Please, send a PR for the doc and I will gladly merge it.

**Comment by jhomarolo** [comment]
@euduardo could you please finish this PR? This is a great contribution!

**Comment by euduardo** [comment]
Created documentation PR: https://github.com/herbsjs/herbsjs.github.io/pull/127

**Comment by jhomarolo** [comment]
Fix #60 as well

**Pull request merged**